app/src/main/java/org/oppia/app/utility/Logger.kt

```diff
@@ -0,0 +1,197 @@
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   */
+  fun w(tag: String, msg: String) {
+    write(LogLevel.WARNING, tag, msg)
+  }
+
+  /**
+   *
+   * @param tag Used to identify the source of a log message.  It usually identifies
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   * @param tr  An exception to log
+   */
+  fun w(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.WARNING, tag, msg, tr)
+  }
+
+  /**
+   *
+   * @param tag Used to identify the source of a log message.  It usually identifies
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   */
+  fun e(tag: String, msg: String) {
+    write(LogLevel.ERROR, tag, msg)
+  }
+
+  /**
+   *
+   * @param tag Used to identify the source of a log message.  It usually identifies
+   * the class or activity where the log call occurs.
+   * @param msg The message you would like logged.
+   * @param tr  An exception to log
+   */
+  fun e(tag: String, msg: String, tr: Throwable) {
+    write(LogLevel.ERROR, tag, msg, tr)
+  }
+
+  private fun isLogEnable(logLevel: LogLevel): Boolean {
+    return GLOBAL_LOG_LEVEL.logLevel < logLevel.logLevel
+  }
+
+  private fun write(logLevel: LogLevel, tag: String, log: String) {
+    if (isLogEnable(logLevel) && ENABLE_CONSOLE_LOG) {
+      Log.println(logLevel.logLevel, tag, log)
+    }
+    if (isLogEnable(logLevel) && ENABLE_FILE_LOG) {
+      val msg = Calendar.getInstance().time.toString() + "\t" + logLevel.name + "/" + tag + ": " + log
+      write(msg)
+    }
+  }
+
+  private fun write(logLevel: LogLevel, tag: String, log: String, tr: Throwable) {
+    if (isLogEnable(logLevel) && ENABLE_CONSOLE_LOG) {
+      Log.println(logLevel.logLevel, tag, log + "\n" + Log.getStackTraceString(tr))
+    }
+    if (isLogEnable(logLevel) && ENABLE_FILE_LOG) {
+      val msg =
+        Calendar.getInstance().time.toString() + "\t" + logLevel.name + "/" + tag + ": " + log + "\n" + Log.getStackTraceString(
+          tr
+        )
+      write(msg)
+    }
+  }
+
+  private fun write(text: String) {
+    var out: BufferedWriter? = null
+    var filePath = LOG_DIRECTORY
+    try {
+
+      val df = SimpleDateFormat("dd_MMM_yyyy", Locale.ENGLISH)
+      val formattedDate = df.format(System.currentTimeMillis())
+
+      if (!File(LOG_DIRECTORY).exists())
+        File(LOG_DIRECTORY).mkdir()
+
+      filePath = "$LOG_DIRECTORY$formattedDate.log"
+      if (!File(filePath).exists())
+        File(filePath).createNewFile()
+
+      val fStream = FileWriter(filePath, true)
+      out = BufferedWriter(fStream)
+      out.write(text + "\n")
+      out.flush()
+    } catch (e: IOException) {
+      Log.e("Log", "Path:$filePath")
+      e.printStackTrace()
+    } finally {
+      try {
+        out?.close()
```

Comment:
Although I suggest using a PrintStream as above, FWIW Kotlin solves this try{}catch{}finally{} logic with ``use``:

```kotlin
FileOutputStream(file).use { stream ->
  stream.write()
}
```

``use`` automatically opens and closes the stream using a try/finally underneath.

Reply:
Made use of printstream so removed above code